duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -371,8 +406,25 @@ public Http build() {
       if (caCerts != null) {
         duoClient.useCustomCertificates(caCerts);
       }
+      if (additionalDuoHeaders != null) {
+        duoClient.addAdditionalDuoHeader(additionalDuoHeaders);
+      }
 
       return duoClient;
     }
+
+
+
+    private void validateXDuoHeader(String name, String value) throws IllegalArgumentException{
+      if (name == null || name.length() == 0){
+        throw new IllegalArgumentException("Not allowed 'Null' or empty header name");
+      } else if (value == null || value.length() == 0){
+        throw new IllegalArgumentException("Not allowed 'Null' or empty header value");
+      } else if (!name.toLowerCase().startsWith("x-duo-")){
```

Comment:
I'm still going to make the argument that we should allow other headers (they just won't get canonicalized)